01-node-tutorial/answers/prompter.js

```diff
@@ -21,18 +23,52 @@ const getBody = (req, callback) => {
 };
 
 // here, you could declare one or more variables to store what comes back from the form.
-let item = "Enter something below.";
+let item = "Talk to Zoltar... Give me a prompt";
+let color = '';
+const defaultSep = path.sep;
+const info = os.userInfo().homedir;
+const re = new RegExp(`([^\\${defaultSep}]+$)`, 'g');
+const userName = info.match(re)[0];
+const machine = os.type();
+let message;
+const colorList = ['lightpink', 'lightcyan', 'palegreen', 'lightgoldenrodyellow', 'pink', 'lightskyblue', 'mediumspringgreen', 'papayawhip', 'lavenderblush', 'paleturquoise', 'mediumseagreen', 'antiquewhite'];
 
 // here, you can change the form below to modify the input fields and what is displayed.
 // This is just ordinary html with string interpolation.
 const form = () => {
+  const uptime = os.uptime();
+  const humanTime = `${parseInt(uptime / 60 / 60)} hours, ${parseInt((uptime / 60)) % 60} minutes ${parseInt(uptime % 60)} seconds`;
+  message = `<br><div>Your name is: <span>${userName}</span>, and you have been working on <span>${machine}</span> for <span>${humanTime}</span> already.<br>
+  <span>Do not forget to take breaks!<br>All the best!</span></div>`;
+
   return `
   <body>
+  <style>
+    body {background-color:${color}}
+    div {font-size:1.3em}
+    span {font-weight:bold}
+  </style>
   <p>${item}</p>
   <form method="POST">
   <input name="item"></input>
+  <label for="colors">Choose the color of the day:</label>
+  <select name="colors" id="colors">
+    <option value="" selected="selected"></option>
+    <option value="${colorList[Math.floor(Math.random() * colorList.length)]}">Green</option>
```

Comment:
This was a very funny and clever take on the assignment, good job!

Reply:
Thank you, it was fun to build it together. Did Zoltar guess your name correctly? Just curious, the code should work on Win PCs but I do not know about Macs.